### PR TITLE
 Fix more closure compiler warnings

### PR DESCRIPTION
**Pull request by sbc100**
This change adds `-sCLOSURE_WARNINGS=error` to our full library
tests and fixes all the resulting closure warnings.

See #12728

### CODE REVIEW

File: src/base64Utils.js [hunk 1]
```diff
@@ -39,8 +39,11 @@ var decodeBase64 = typeof atob === 'function' ? atob : function (input) {
   return output;
 };
 
-// Converts a string of base64 into a byte array.
-// Throws error on invalid input.
+/**
+ * Converts a string of base64 into a byte array.
+ * Throws error on invalid input.
+ * @param {string} s
```
Comment (kripken) [inline src/base64Utils.js:45]: what warning does this avoid?

File: tests/common.py [hunk 2]
```diff
@@ -405,6 +405,10 @@ def setUp(self):
     super().setUp()
     self.settings_mods = {}
     self.emcc_args = ['-Werror']
+    # We want to be strict about closure warnings in our test code.
+    # TODO(sbc): Remove this if we make it the default for `-Werror`:
+    # https://github.com/emscripten-core/emscripten/issues/16205):
+    self.ldflags = ['-sCLOSURE_WARNINGS=error']
```
Comment (kripken) [inline tests/common.py:411]: Isn't this the same as `IGNORE_CLOSURE_COMPILER_ERRORS` which is default to off? I actually don't remember when we added `CLOSURE_WARNINGS`. Maybe I'm not understanding how the two differ?

File: tests/common.py [hunk 3]
```diff
@@ -827,7 +833,7 @@ def get_library(self, name, generated_libs, configure=['sh', './configure'],
     build_dir = self.get_build_dir()
     output_dir = self.get_dir()
 
-    emcc_args = self.get_emcc_args()
+    emcc_args = self.get_emcc_args(ldflags=False)
```
Comment (kripken) [inline tests/common.py:836]: why don't we want ldflags here? a comment would be good.

File: tests/test_core.py [hunk 4]
```diff
@@ -7185,6 +7185,7 @@ def test_embind_no_rtti_followed_by_rtti(self):
   })
   def test_webidl(self, mode, allow_memory_growth):
     self.uses_es6 = True
+    self.ldflags.remove('-sCLOSURE_WARNINGS=error')
```
Comment (kripken) [inline tests/test_core.py:7188]: maybe a comment with a TODO to remove this at the proper time?

File: tests/common.py (hunk 2 above)
Comment (sbc100) [reply tests/common.py:411]: `IGNORE_CLOSURE_COMPILER_ERRORS` I believe controls how closer handles what it considers errors, whereas  `CLOSURE_WARNINGS` control what it only considers to be warnings. 

I also think `IGNORE_CLOSURE_COMPILER_ERRORS` pre-dates `CLOSURE_WARNINGS` (and I'm not quite sure what happens if you ignore errors, or when you might want to).

In any case this change is about warnings.. errors are already fatal by default.

**Comment by sbc100** [comment]
Adding @kainino0x for webgpu changes.. in particular I'm not sure about the `Fence` change.. its just fixed the warning.

File: src/library_webgpu.js [hunk 5]
```diff
@@ -192,6 +191,7 @@ var LibraryWebGPU = {
       {{{ gpu.makeInitManager('Device') }}}
       {{{ gpu.makeInitManager('Queue') }}}
 
+      {{{ gpu.makeInitManager('Fence') }}}
```
Comment (kainino0x) [inline src/library_webgpu.js:194]: Fences have been removed. You can leave this line out, and instead remove whatever was depending on it - looks like:
https://github.com/emscripten-core/emscripten/blob/main/src/library_html5_webgpu.js#L74

Other WebGPU changes LGTM

Comment (sbc100) [reply src/library_webgpu.js]: Done!  Thanks.

File: src/base64Utils.js (hunk 1 above)
Comment (sbc100) [reply src/base64Utils.js]: Reverted.

**Comment by sbc100** [comment]
I reduced the scope of this PR so now it only includes enough to run other.test_closure_full_js_library and other.test_closure_webgpu without any warnings.

**Pull request merged**